x-pack/legacy/plugins/file_upload/public/util/file_parser.test.js

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { parseAndClean, jsonPreview } from './file_parser';
+
+describe('parse file', () => {
+  const cleanAndValidate = jest.fn(a => a);
+  const resolve = jest.fn();
+  const reject = jest.fn();
+  const previewFunction = jest.fn();
+
+  beforeEach(() => {
+    cleanAndValidate.mockClear();
+    resolve.mockClear();
+    reject.mockClear();
+    previewFunction.mockClear();
+  });
+
+  it('should parse valid JSON', () => {
+    const validJsonFileResult = {
+      target: {
+        result: JSON.stringify({
+          'type': 'Feature',
+          'geometry': {
+            'type': 'Polygon',
+            'coordinates': [[
+              [-104.05, 78.99],
+              [-87.22, 78.98],
+              [-86.58, 75.94],
+              [-104.03, 75.94],
+              [-104.05, 78.99]
+            ]]
+          },
+        })
+      }
+    };
+
+    parseAndClean(cleanAndValidate, resolve, reject)(validJsonFileResult);
+    // Confirm cleanAndValidate called
+    expect(cleanAndValidate.mock.calls.length).toEqual(1);
+    // Confirm resolve called
+    expect(resolve.mock.calls.length).toEqual(1);
+  });
+
+  it('should reject invalid JSON', () => {
+    const inValidJsonFileResult = {
+      target: {
+        result: 'this should not work'
+      }
+    };
+
+    parseAndClean(cleanAndValidate, resolve, reject)(inValidJsonFileResult);
+    // Confirm cleanAndValidate not called
+    expect(cleanAndValidate.mock.calls.length).toEqual(0);
+    // Confirm resolve not called
+    expect(resolve.mock.calls.length).toEqual(0);
+    // Confirm reject called
+    expect(reject.mock.calls.length).toEqual(1);
+  });
+
+  it('should reject object instead of JSON', () => {
```

Comment:
interesting. why this restriction?

Reply:
It was something I accidentally did at one point (pass in an already parsed object). There's enough parsing and transformation happening that it was easier than you might think to call it with the wrong data while reorganizing. I could also remove it. No strong feelings either way.

Reply:
Actually, I've removed it!